lib/web_of_science/process_records.rb

```diff
@@ -43,12 +44,21 @@ def create_publications
       ## 1
       # Filter and select new WebOfScienceSourceRecords
       # @return [Array<WebOfScience::Record>]
-      def match_wos_records
+      def filter_databases
         return [] if records.count.zero?
-        records.select { |rec| WebOfScienceSourceRecord.where(uid: rec.uid).count.zero? }
+        return records if Settings.WOS.ACCEPTED_DBS.empty?
+        records.select { |rec| Settings.WOS.ACCEPTED_DBS.include? rec.database }
       end
 
       ## 2
+      # Filter and select new WebOfScienceSourceRecords
+      # @return [Array<WebOfScience::Record>]
+      def match_wos_records(records)
+        return [] if records.count.zero?
+        records.select { |rec| WebOfScienceSourceRecord.where(uid: rec.uid).count.zero? }
```

Comment:
For 100 records, these does 100 queries on `WebOfScienceSourceRecord`'s table when it could just do one.

Reply:
Yes, that change could be made, although it wasn't the focus of this PR.  I'll go ahead and try to resolve that anyway in this PR.